src/sagemaker/workflow/pipeline.py

```diff
@@ -744,6 +738,20 @@ def upsert(
         raise RuntimeError("Immutable Pipelines cannot be updated")
 
 
+@attr.s
+class PipelineList:
```

Comment:
I guess users won't initiate an object for it directly. Let's make it an internal class: `_PipelineList`